packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SVGComponents/SVGPanZoom/SVGPanZoom.js

```diff
@@ -0,0 +1,145 @@
+}) {
+  const scrollContainer = createRef()
+  const defaultViewBox = {
+    x: 0,
+    y: 0,
+    height: naturalHeight,
+    width: naturalWidth
+  }
+
+  const [ scale, setScale ] = useState(1)
+  const [ subjectScale, setSubjectScale ] = useState(1)
+  const [ viewBox, setViewBox ] = useState(defaultViewBox)
+
+  function onMount () {
+    setOnDrag(onDrag)
+    setOnPan(onPan)
+    setOnZoom(onZoom)
+    scrollContainer.current.addEventListener('wheel', e => e.preventDefault())
+  }
+
+  function onUnmount () {
+    setOnDrag(() => true)
+    setOnPan(() => true)
+    setOnZoom(() => true)
+    scrollContainer.current.removeEventListener('wheel', e => e.preventDefault())
+  }
+
+  useEffect(() => {
+    onMount()
+    return onUnmount
+  }, [])
+
+  useEffect(function updateImageSize () {
+    onImageChange(img)
+  }, [img])
+
+  useEffect(function onScaleChange () {
+    const newViewBox = scaledViewBox(scale)
+    setViewBox(newViewBox)
+  }, [scale])
+
+  function onImageChange (img) {
+    const { width: clientWidth, height: clientHeight } = img ? img.getBoundingClientRect() : {}
+    const subjectScale = clientWidth / naturalWidth
+    setSubjectScale(subjectScale)
+  }
+
+  function scaledViewBox (scale) {
+    const viewBoxScale = 1 / scale
+    const xCentre = viewBox.x + viewBox.width / 2
+    const yCentre = viewBox.y + viewBox.height / 2
+    const width = parseInt(naturalWidth * viewBoxScale, 10)
+    const height = parseInt(naturalHeight * viewBoxScale, 10)
+    const x = xCentre - width / 2
+    const y = yCentre - height / 2
+    return { x, y, width, height }
+  }
+
+  function onDrag (event, difference) {
+    setViewBox(prevViewBox => {
+      const newViewBox = Object.assign({}, prevViewBox)
+      newViewBox.x -= difference.x / 1.5
+      newViewBox.y -= difference.y / 1.5
+      return newViewBox
+    })
+  }
+
+  function onPan (dx, dy) {
+    setViewBox(prevViewBox => {
+      const newViewBox = Object.assign({}, prevViewBox)
+      newViewBox.x -= dx * 10
+      newViewBox.y += dy * 10
+      return newViewBox
+    })
+  }
+
+  function onZoom (type) {
+    switch (type) {
+      case 'zoomin': {
+        setScale(prevScale => Math.min(prevScale + 0.1, maxZoom))
+        onImageChange(img)
+        return
+      }
+      case 'zoomout': {
+        setScale(prevScale => Math.max(prevScale - 0.1, minZoom))
+        onImageChange(img)
+        return
+      }
+      case 'zoomto': {
+        setScale(1)
+        setViewBox({
+          x: 0,
+          y: 0,
+          width: naturalWidth,
+          height: naturalHeight
+        })
+      }
+    }
+  }
+
+  function onWheel (event) {
+    const { deltaY } = event
+    if (deltaY < 0) {
+      onZoom('zoomout', -1)
+    } else {
+      onZoom('zoomin', 1)
+    }
+  }
+
+  return (
+    <div ref={scrollContainer} onWheel={onWheel}>
+      {cloneElement(children, { scale: subjectScale, viewBox: `${viewBox.x} ${viewBox.y} ${viewBox.width} ${viewBox.height}` })}
```

Comment:
This should definitely be `subjectScale`. Here's the corresponding lines on master.
https://github.com/zooniverse/front-end-monorepo/blob/555b5688314eae35612802c9f27a53c201b62d73/packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageViewerContainer.js#L224-L225

I'm seeing `Nan` errors that stop the SVG from rendering, since rebasing this to the latest master, so passing the value of either `clientWidth` or `naturalWidth` must have changed in a recent merge.

Reply:
This is ironic considering that this PR fixes a bug where the SVG renders before `naturalWidth` is known, leading to `NaN` errors.

Reply:
Something I could do in this PR is rename variables to make clear the distinction between image scale (rendered size / actual size) and zoom level (starts at 1 and goes up or down in steps of .1 as we zoom in and out.)

Reply:
Fixed in the code now. `clientWidth` wasn't being updated after the subject image re-renders. Tests are failing so I'll take a look at that.

Reply:
Tests should be fixed too.